### DISCUSSION TITLE
InfluxDB integration into scripting

### DISCUSSION THREAD

**Discussion by pkkrusty** [arendst/Tasmota#15734] [Q&A] [answered]
What's the best method to get variables and other data from a script to InfluxDB? I can do a bunch of webquerys, but maybe there's a way the driver is already integrated if I'm pushing json out?

**Comment by barbudor** [comment]
I still owe you a `IfxWrite` is you are still interrested

The current driver is pushing to InfluxDB all the content of the SENSOR message, with each field on a separated measurment/bucket
Which is not what I like and this  is why I'm not using the driver but doing it from Node-RED.
I'm writing to InfluxDB a single record per device with all datas
So instead of having a bucket for temperatire, one for humidity etc .... I have a sensor bucket with all values

In the screenshot below, `name` is a key and all others are values

![image](https://user-images.githubusercontent.com/2996042/171728492-20ae927d-7848-4558-b1e9-9a6a64cde8f8.png)

This is done by a single write per device with `sensor,name=xxxxx temp=xxx,hum=xxxx` 
or for the vindriktning+CCS811, it's `sensor,name=sns-envsalon pm2_5=xxxx,eco2=xxxx,tvoc=xxxx`
So, as you can see, I'm mixing in the same bucket sensors that have different characteristics which is not a problem
And when I retrieve a record, I have all measurments form that sensor, not just a single value

Which means that you don't need "a bunch of webquerys" if you decide to organize your data this way
A single one can push all of your data

**Comment by pkkrusty** [comment]
Definitely still interested. Seems like a better option, and cuts down on chars used in script and rules.

**Reply by barbudor** [reply]
Rainy week end ahead, should get more time for coding 😂

**Comment by pkkrusty** [comment, marked as answer]
Turns out that pushing the >J section in scripting publishes to SENSOR and the InfluxDB driver picks up on that like a normal SENSOR publishes. So one can just push >J at whatever interval you want and get your InfluxDB integration. No webqueries necessary. Nice!
```
>J
,"Battery":{
"Cell-1":%cv[1]%,
"Cell-2":%cv[2]%,
"Cell-3":%cv[3]%,
"Cell-4":%cv[4]%,
"Cell-5":%cv[5]%,
"Cell-6":%cv[6]%,
"Cell-7":%cv[7]%,
"Cell-8":%cv[8]%,
"Cell-9":%cv[9]%,
"Cell-10":%cv[10]%,
"Cell-11":%cv[11]%,
"Cell-12":%cv[12]%,
"Max":%max%,
"Min":%min%
}
```
produces
```
11:04:44.474 MQT: pkkrusty_AFAC99/tele/SENSOR = {"Time":"2022-06-04T11:04:44","Battery":{"Cell-1":3.81,"Cell-2":3.74,"Cell-3":3.69,"Cell-4":3.72,"Cell-5":3.72,"Cell-6":3.75,"Cell-7":3.72,"Cell-8":3.74,"Cell-9":3.69,"Cell-10":3.72,"Cell-11":3.72,"Cell-12":3.75,"Max":3.81,"Min":3.69}}
11:19:44.452 MQT: pkkrusty_AFAC99/tele/STATE = {"Time":"2022-06-04T11:19:44","Uptime":"0T04:15:10","UptimeSec":15310,"Heap":9,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":2,"SSId":"Roatan-2","BSSId":"C0:4A:00:40:5F:E7","Channel":6,"Mode":"11n","RSSI":72,"Signal":-64,"LinkCount":1,"Downtime":"0T00:00:04"}}
11:19:44.459 MQT: pkkrusty_AFAC99/tele/SENSOR = {"Time":"2022-06-04T11:19:44","Battery":{"Cell-1":3.81,"Cell-2":3.74,"Cell-3":3.69,"Cell-4":3.72,"Cell-5":3.72,"Cell-6":3.75,"Cell-7":3.72,"Cell-8":3.74,"Cell-9":3.69,"Cell-10":3.72,"Cell-11":3.72,"Cell-12":3.75,"Max":3.81,"Min":3.69}}
```
So MQTT is publishing on it's set 15 minute interval, but InfluxDB driver is sending on a 60 second interval as defined by IFXPeriod.
<img width="843" alt="Screen Shot 2022-06-04 at 11 36 59" src="https://user-images.githubusercontent.com/79770016/171997284-d06e623c-193f-43e9-ab6e-67b386a09b80.png">

@gemu2015 Any details to add?

**Reply by gemu2015** [reply]
@pkkrusty 
nice you solved your BMS SPI access!
 do not use Influx but glad that it was that easy to get it to work.

**Reply by pkkrusty** [reply]
In many of your examples you use the median m: instead of the mean M: to store the arrays. For my purposes I'd like to use mean cell voltage to compare for activating cell balancing. Any difference in array usage using M: instead of m:?

**Reply by gemu2015** [reply]
no diference when using as an array.

to use a mean filter define M filter with desired filter length (default is 8)
then simply store new values into the array (without index) e.g every second.
then read the array (without index) and you will get the moving average of 8 values.
